task_processing/plugins/mesos/execution_framework.py

```diff
@@ -138,8 +161,28 @@ def kill_tasks_stuck_in_staging(self):
                         )
                         get_metric(TASK_STUCK_COUNT).count(1)
 
+            self._reconcile_tasks()
             time.sleep(10)
 
+    def _reconcile_tasks(self):
+        if time.time() < self._reconcile_tasks_at:
+            return
+
+        tasks_to_reconcile = [Dict({'task_id': task_id}) for
+                              task_id in self.task_metadata]
+
+        log.info('Reconciling following tasks {tasks}'.format(
+            tasks=tasks_to_reconcile
+        ))
+
+        if len(tasks_to_reconcile) > 0:
```

Comment:
I don't think you need this - it's fine to reconcile with an empty list (and ask mesos for a status update on *every* task)

Reply:
Pymesos is not handling it correctly, that's why we have to do it. https://github.com/douban/pymesos/blob/cd6569cc5d167a8edf819c569a2c61dd05e4053e/pymesos/scheduler.py#L316